include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -0,0 +1,1523 @@
+        assert(path.has_value());
+
+        route_path(layout, *path);
+
+        for (const auto& el : *path)
+        {
+            layout.obstruct_coordinate(el);
+        }
+    }
+    /**
+     * Places a node with two inputs in the layout and routes it.
+     *
+     * @param position The tile representing the position for placement.
+     * @param layout The layout in which to place the node.
+     * @param node2pos A dictionary mapping nodes from the network to signals in the layout.
+     * @param fc A vector of nodes that precede the double fanin node.
+     */
+    void route_double_input_node(const tile<ObstrLyt>& position, ObstrLyt& layout,
+                                 node_dict_type<ObstrLyt, tec_nt>& node2pos, const fanin_container<tec_nt>& fc) noexcept
+    {
+        const auto& pre1 = fc.fanin_nodes[0];
+        const auto& pre2 = fc.fanin_nodes[1];
+
+        const auto pre1_t = static_cast<tile<ObstrLyt>>(node2pos[pre1]);
+        const auto pre2_t = static_cast<tile<ObstrLyt>>(node2pos[pre2]);
+
+        layout.move_node(layout.get_node(position), position, {});
+
+        const auto path_1 = check_path(layout, pre1_t, position, false, false);
+        assert(path_1.has_value());
+
+        for (const auto& el : *path_1)
+        {
+            layout.obstruct_coordinate(el);
+        }
+
+        const auto path_2 = check_path(layout, pre2_t, position, false, false);
+        assert(path_2.has_value());
+
+        for (const auto& el : *path_2)
+        {
+            layout.obstruct_coordinate(el);
+        }
+
+        route_path(layout, *path_1);
+        route_path(layout, *path_2);
+    }
+    /**
+     * Executes a single placement step in the layout for the given network node. It determines the type of the node,
+     * places it accordingly, and checks if a solution was found.
+     *
+     * @param position The tile representing the position for placement.
+     * @param layout The layout in which to place the node.
+     * @param network The network containing the nodes.
+     * @param current_node The current node index in the nodes_to_place vector.
+     * @param nodes_to_place A vector representing the nodes to be placed.
+     * @param current_po The current primary output index.
+     * @param node2pos A dictionary mapping nodes from the network to signals in the layout.
+     * @param pi2node A mapping of primary input nodes to layout nodes.
+     * @return A boolean indicating if a solution was found.
+     */
+    [[nodiscard]] bool place_and_route(const tile<ObstrLyt>& position, ObstrLyt& layout, const tec_nt& network,
+                                       uint64_t& current_node, std::vector<mockturtle::node<tec_nt>>& nodes_to_place,
+                                       uint64_t& current_po, node_dict_type<ObstrLyt, tec_nt>& node2pos,
+                                       mockturtle::node_map<mockturtle::node<ObstrLyt>, tec_nt>& pi2node) noexcept
```

Comment:
Please double-check this function's parameters. We are passing some parameters here that the function can (and will) change. This usually indicates that these parameters should rather be members of some class. Can that be implemented here?